integration-tests/common/test_common.go

```diff
@@ -0,0 +1,214 @@
+	ctfClient "github.com/smartcontractkit/chainlink-testing-framework/client"
+	"github.com/smartcontractkit/chainlink/integration-tests/client"
+)
+
+const (
+	ChainName = "starknet"
+	ChainId   = gateway.GOERLI_ID // default chainID for local devnet and live testnet
+	// These are one of the default addresses based on the seed we pass to devnet which is 123
+	defaultWalletAddress = "0x6e3205f9b7c4328f00f718fdecf56ab31acfb3cd6ffeb999dcbac41236ea502"
+	defaultWalletPrivKey = "0xc4da537c1651ddae44867db30d67b366"
+)
+
+var (
+	err       error
+	clImage   string
+	clVersion string
+)
+
+func init() {
+	// pass in flags to override default chainlink image & version
+	flag.StringVar(&clImage, "chainlink-image", "", "specify chainlink image to be used")
+	flag.StringVar(&clVersion, "chainlink-version", "", "specify chainlink version to be used")
+}
+
+type Test struct {
+	sc         *StarkNetDevnetClient
+	cc         *ChainlinkClient
+	mockServer *ctfClient.MockserverClient
+	Env        *environment.Environment
+	Common     *Common
+}
+
+type StarkNetDevnetClient struct {
+	ctx    context.Context
+	cancel context.CancelFunc
+	cfg    *blockchain.EVMNetwork
+	client *resty.Client
+}
+type ChainlinkClient struct {
+	nKeys          []ctfClient.NodeKeysBundle
+	chainlinkNodes []*client.Chainlink
+	bTypeAttr      *client.BridgeTypeAttributes
+	bootstrapPeers []client.P2PData
+}
+
+// DeployCluster Deploys and sets up config of the environment and nodes
+func (t *Test) DeployCluster(nodes int, commonConfig *Common) {
+	t.Common = SetConfig(commonConfig)
+	t.cc = &ChainlinkClient{}
+	t.sc = &StarkNetDevnetClient{}
+	t.DeployEnv(nodes)
+	t.SetupClients()
+	t.cc.nKeys, t.cc.chainlinkNodes, err = t.Common.CreateKeys(t.Env)
+	Expect(err).ShouldNot(HaveOccurred(), "Creating chains and keys should not fail")
+}
+
+// DeployEnv Deploys the environment
+func (t *Test) DeployEnv(nodes int) {
+	clConfig := map[string]interface{}{
+		"replicas": nodes,
+		"env": map[string]interface{}{
+			"STARKNET_ENABLED":            "true",
+			"EVM_ENABLED":                 "false",
+			"EVM_RPC_ENABLED":             "false",
+			"CHAINLINK_DEV":               "false",
+			"FEATURE_OFFCHAIN_REPORTING2": "true",
+			"feature_offchain_reporting":  "false",
+			"P2P_NETWORKING_STACK":        "V2",
+			"P2PV2_LISTEN_ADDRESSES":      "0.0.0.0:6690",
+			"P2PV2_DELTA_DIAL":            "5s",
+			"P2PV2_DELTA_RECONCILE":       "5s",
+			"p2p_listen_port":             "0",
+		},
+	}
+
+	// if image is specified, include in config data
+	// if not, do not set image data - will default to env vars
+	if clImage != "" && clVersion != "" {
+		clConfig["chainlink"] = map[string]interface{}{
+			"image": map[string]interface{}{
+				"image":   clImage,
+				"version": clVersion,
+			},
+		}
+	}
+
+	t.Env = environment.New(&environment.Config{
+		NamespacePrefix: "chainlink-smoke-ocr-starknet-ci",
+		InsideK8s:       false,
+	}).
+		//AddHelm(hardhat.New(nil)).
```

Comment:
TODO: reenable with lower resource usage

Reply:
update: can't be re-enabled yet b/c in local testing it never shows a ready state